Data Source: https://www.forbesindia.com/article/explainers/nifty-50-history/93195/1

The **NIFTY50** is a benchmark stock market index in India, representing the weighted average of the 50 largest companies listed on the National Stock Exchange (NSE). It is widely regarded as an indicator of the performance of the Indian equity market and includes companies from various sectors. The NIFTY50 is used by investors to gauge the overall health of the market and to make informed investment decisions based on its movements.


**Objective of the Code:**

The objective of this code is to analyze the historical performance of the NIFTY50 index by calculating the Compound Annual Growth Rate (CAGR) over various investment horizons (1 year, 2 years, etc.). The code aims to assess the likelihood of positive and negative returns over different periods by computing the percentage of instances with positive and negative CAGR values. This analysis helps investors understand the potential risks and returns associated with different investment durations in the NIFTY50 index.

In [1]:
import pandas as pd

# Historical data from the user input
data = {
    "Year": [1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023, 2024],
    "Closing Value": [330.86, 558.63, 761.31, 1042.5, 1182.2, 908.53, 899.1, 1079.4, 884.25, 1480.40, 1263.50, 1059, 1093.50, 1879.70, 2080.50, 2836.50, 3966.40, 6138.60, 2959.10, 5201, 6134.50, 4624.30, 5905.10, 6304, 8282.70, 7946.30, 8185.80, 10530, 10862.55, 12168.45, 13981.75, 17354.05, 18105.30, 21731.40, 25004.00]
}

# Convert to DataFrame
df = pd.DataFrame(data)

# Function to calculate CAGR
def calculate_cagr(start_value, end_value, num_years):
    return (end_value / start_value) ** (1 / num_years) - 1

# Calculate rolling CAGR for different horizons (1 year, 2 years, etc.)
cagr_results = []

for horizon in [1, 2, 3, 5, 7, 10]:
    cagr_values = []
    for i in range(len(df) - horizon):
        start_value = df.loc[i, "Closing Value"]
        end_value = df.loc[i + horizon, "Closing Value"]
        num_years = horizon
        cagr = calculate_cagr(start_value, end_value, num_years)
        cagr_values.append(cagr)
    cagr_results.append({
        "Investment Period": horizon,
        "Negative CAGR %": sum([1 for x in cagr_values if x < 0]) / len(cagr_values) * 100,
        "Positive CAGR %": sum([1 for x in cagr_values if x >= 0]) / len(cagr_values) * 100,
       })

# Convert results to DataFrame for display
cagr_df = pd.DataFrame(cagr_results)

# Display the result
cagr_df


Investment Period  Negative CAGR %  Positive CAGR %
0                  1        23.529412        76.470588
1                  2        30.303030        69.696970
2                  3        15.625000        84.375000
3                  5         6.666667        93.333333
4                  7         3.571429        96.428571
5                 10         0.000000       100.000000

In [2]:
# Function to calculate CAGR
def calculate_cagr(start_value, end_value, num_years):
    return (end_value / start_value) ** (1 / num_years) - 1

# Calculate 1-Year CAGR values
cagr_1_year = [None]  # Start with None for 1990 since there's no previous year to compare

for i in range(1, len(df)):
    start_value = df.loc[i-1, "Closing Value"]
    end_value = df.loc[i, "Closing Value"]
    num_years = 1
    cagr = calculate_cagr(start_value, end_value, num_years)
    cagr_1_year.append(cagr)

# Add the 1-Year CAGR values to the DataFrame
df["1-Year CAGR"] = pd.Series(cagr_1_year)

# Calculate the percentage of negative CAGR values
negative_cagr_percentage = sum(df["1-Year CAGR"].dropna() < 0) / len(df["1-Year CAGR"].dropna()) * 100

In [3]:
print(df)

    Year  Closing Value  1-Year CAGR
0   1990         330.86          NaN
1   1991         558.63     0.688418
2   1992         761.31     0.362816
3   1993        1042.50     0.369350
4   1994        1182.20     0.134005
5   1995         908.53    -0.231492
6   1996         899.10    -0.010379
7   1997        1079.40     0.200534
8   1998         884.25    -0.180795
9   1999        1480.40     0.674187
10  2000        1263.50    -0.146514
11  2001        1059.00    -0.161852
12  2002        1093.50     0.032578
13  2003        1879.70     0.718976
14  2004        2080.50     0.106826
15  2005        2836.50     0.363374
16  2006        3966.40     0.398343
17  2007        6138.60     0.547650
18  2008        2959.10    -0.517952
19  2009        5201.00     0.757629
20  2010        6134.50     0.179485
21  2011        4624.30    -0.246181
22  2012        5905.10     0.276972
23  2013        6304.00     0.067552
24  2014        8282.70     0.313880
25  2015        7946.30    -0.040615
2

### Final Conclusion:
The analysis of the NIFTY50 index over different investment periods reveals the following trends:

Short-term Investments (1-3 years):

- There is a relatively higher risk of negative returns, with 23.53% of 1-year periods, 30.30% of 2-year periods, and 15.63% of 3-year periods experiencing negative CAGR.

Medium-term Investments (5-7 years):

- The risk of negative returns decreases significantly, with only 6.67% of 5-year periods and 3.57% of 7-year periods showing negative CAGR.

Investing for at least 10 years in the NIFTY50 index is shown to be the most secure, with 0% of these periods resulting in negative returns.
- Every single 10-year investment period in the dataset resulted in positive returns, with 100% of these periods showing positive CAGR.